addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -295,13 +295,21 @@ odoo.define('website.s_website_form', function (require) {
             // force server date format usage for existing fields
             this.$target.find('.s_website_form_field:not(.s_website_form_custom)')
             .find('.s_website_form_date, .s_website_form_datetime').each(function () {
-                var date = $(this).datetimepicker('viewDate').clone().locale('en');
-                var format = 'YYYY-MM-DD';
-                if ($(this).hasClass('s_website_form_datetime')) {
+                const inputEl = this.querySelector('input');
+
+                // Datetimepicker('viewDate') will return `new Date()` if the
+                // input is empty but we want to keep the empty value
+                if (!inputEl.value) {
+                    return;
+                }
+
+                let date = $(this).datetimepicker('viewDate').clone().locale('en');
+                let format = 'YYYY-MM-DD';
+                if (this.classList.contains('s_website_form_datetime')) {
```

Comment:
Don't adapt lines not related to your diff, this is useless change and will conflict in forward port if we fix this line one day, on top of being another source of bug (the more diff the more risk)
When you touch a line for a real diff, it's worth rewriting it as the risk is already taken (you modified it) and the history will be changed anyway and will prevent commit to be ported automatically.